crates/holochain_websocket/src/lib.rs

```diff
@@ -688,13 +729,59 @@ impl WebsocketListener {
         Ok(Self {
             config,
             access_control,
-            listener,
+            listener: Box::new(listener),
+        })
+    }
+
+    /// Bind a new websocket listener on the same port using a v4 and a v6 socket.
+    pub async fn dual_bind(
+        config: Arc<WebsocketConfig>,
+        addr_v4: impl Into<SocketAddr>,
+        addr_v6: impl Into<SocketAddr>,
+    ) -> Result<Self> {
+        let access_control = Arc::new(config.allowed_origins.clone().ok_or_else(|| {
+            Error::other("WebsocketListener requires access control to be set in the config")
+        })?);
+
+        let mut addr_v4 = addr_v4.into();
+        let addr_v6 = addr_v6.into();
+
+        if !addr_v4.is_ipv4() {
+            return Err(Error::other("Expected an IPv4 address"));
+        }
+
+        if !addr_v6.is_ipv6() {
+            return Err(Error::other("Expected an IPv6 address"));
+        }
```

Comment:
you could have the function signature take [SocketAddrV4](https://doc.rust-lang.org/std/net/struct.SocketAddrV4.html) and [SocketAddrV6](https://doc.rust-lang.org/std/net/struct.SocketAddrV6.html) instead.